src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -448,20 +452,59 @@ void ReberGrammarTestNetwork(bool embedded = false)
   BOOST_REQUIRE_GE(successes, 1);
 }
 
+BOOST_AUTO_TEST_CASE(GRUForwardTest)
+{
+  arma::Mat<char> transitions;
+  transitions << 'T' << 'P' << '1' << '2' << arma::endr
+              << 'X' << 'S' << '3' << '1' << arma::endr
+              << 'V' << 'T' << '4' << '2' << arma::endr
+              << 'X' << 'S' << '2' << '5' << arma::endr
+              << 'P' << 'V' << '3' << '5' << arma::endr
+              << 'E' << 'E' << '0' << '0' << arma::endr;
+
+  std::string testReber = "BPTVVE";
+  arma::mat testInput;
+  arma::colvec translation;
+  
+  for (size_t j = 0; j < testReber.length() - 1; j++)
+  {
+    ReberTranslation(testReber[j], translation);
+    testInput = arma::join_cols(testInput, translation);
+  }
+  
+  GRU<> gru(14, 7, 4);
+}
+
+/**
+ * Train the specified networks on a Reber grammar dataset.
+ */
+BOOST_AUTO_TEST_CASE(LSTMReberGrammarTest)
+{
+  ReberGrammarTestNetwork<LSTM<>>(false);
+}
+
+/**
+ * Train the specified networks on an embedded Reber grammar dataset.
+ */
+BOOST_AUTO_TEST_CASE(LSTMEmbeddedReberGrammarTest)
+{
+  ReberGrammarTestNetwork<LSTM<>>(true);
+}
+
 /**
  * Train the specified networks on a Reber grammar dataset.
  */
-BOOST_AUTO_TEST_CASE(ReberGrammarTest)
+BOOST_AUTO_TEST_CASE(GRUReberGrammarTest)
```

Comment:
I end up with ``error: Mat::rows(): indices out of bounds or incorrectly used`` running ``RecurrentNetworkTest/GRUReberGrammarTest`` wonder if you missed to check in something crucial?